src/simulation.py

```diff
@@ -0,0 +1,233 @@
+from src.custom_types import Solution, Vertex, RGBA, Canvas
+from src.reconstruction import polygon_init, polygon_mutate
+from src.visualize import add_polygon, visualize_canvas
+from src.loss import sad
+from numpy import array
+from copy import copy, deepcopy
+import numpy as np
+
+
+class Simulation:
+    def __init__(self, **kwargs):
+        """
+        Init simulation.
+        Keywords:
+            - Base image
+            - Output image
+            - Max generations
+            - Max iterations
+            - Stagnation limit
+            - Number evaluations
+            - Height
+            - Width
+            - Number Verticies
+        """
+
+        self.base_image = kwargs.get("b_image")
+        self.output_image = kwargs.get("o_image")
+        self.max_generations: int = kwargs.get("m_gen", 10)
+        self.max_iterations: int = kwargs.get("m_iter", 10)
+        self.stagnation_limit: int = kwargs.get("stag_lim", 10)
+        self.n_verticies: int = kwargs.get("n_vert", 3)
+        self.num_evals: int = kwargs.get("n_evals", 50000)
+        # NOTE: this value is from the paper
+
+        # Derived class variables
+        self.height, self.width = self.base_image.shape
+        self.canvas = Canvas(list())
+        self.counter = 0
+
+        self.canvas = self.create_polygon(self.canvas)
+
+    def update_probabilities(
```

Comment:
If we have this, we might want to get rid of `mut_prob` method in `Canvas` class.